### PR TITLE
add tests for Data Model coverage

### PR DESCRIPTION
**Pull request by jaeopt**

### CODE REVIEW

File: OptimizelySDK/Data Model/Audience/AttributeValue.swift [hunk 1]
```diff
@@ -81,10 +81,13 @@ enum AttributeValue: Codable, Equatable, CustomStringConvertible {
             return
         }
         
-        if let value = try? container.decode(Int64.self) {
+        // Int values are all filtered as Double, so this Int parsing is not needed here
```
Comment (thomaszurkan-optimizely) [inline OptimizelySDK/Data Model/Audience/AttributeValue.swift]: remove the comments and commented out code.

File: OptimizelySDK/Data Model/Audience/ConditionHolder.swift [hunk 2]
```diff
@@ -95,9 +95,6 @@ extension Array where Element == ConditionHolder {
     }
     
     func evaluate(op: LogicalOp, project: ProjectProtocol?, attributes: OptimizelyAttributes?) throws -> Bool {
-        guard self.count > 0 else {
```
Comment (thomaszurkan-optimizely) [inline OptimizelySDK/Data Model/Audience/ConditionHolder.swift]: why is this removed?

File: OptimizelySDK/Data Model/Audience/UserAttribute.swift [hunk 3]
```diff
@@ -97,7 +97,7 @@ extension UserAttribute {
 
         // invalid match - parsed for forward compatibility only (but evaluation fails)
         guard let matchFinal = matchSupported else {
-            throw OptimizelyError.userAttributeInvalidMatch(self.match ?? "empty")
+            throw OptimizelyError.userAttributeInvalidMatch(self.match!)
```
Comment (thomaszurkan-optimizely) [inline OptimizelySDK/Data Model/Audience/UserAttribute.swift]: there is no guarantee that match is not nil.  so either bind it or default it but not bang.

File: OptimizelySDK/Data Model/DispatchEvents/EventForDispatch.swift [hunk 4]
```diff
@@ -35,7 +35,13 @@ import Foundation
 }
 
 extension EventForDispatch {
+    
     override public var description: String {
         return "[url] \(url) (" + (String(data: body, encoding: .utf8) ?? "UNKNOWN") + ")"
     }
+
```
Comment (thomaszurkan-optimizely) [inline OptimizelySDK/Data Model/DispatchEvents/EventForDispatch.swift]: why did you move this?

File: OptimizelySDK/OptimizelyTests/OptimizelyTests-DataModel/ConditionLeaf.swift [hunk 5]
```diff
@@ -0,0 +1,23 @@
+/****************************************************************************
```
Comment (thomaszurkan-optimizely) [inline OptimizelySDK/OptimizelyTests/OptimizelyTests-DataModel/ConditionLeaf.swift]: why is this class here?

File: OptimizelySDK/Data Model/Audience/ConditionHolder.swift (hunk 2 above)
Comment (jaeopt) [reply OptimizelySDK/Data Model/Audience/ConditionHolder.swift]: This condition is covered by the caller, so the guard-else condition never happens

File: OptimizelySDK/Data Model/Audience/UserAttribute.swift (hunk 3 above)
Comment (jaeopt) [reply OptimizelySDK/Data Model/Audience/UserAttribute.swift]: nil match case is filtered out by "matchSupported", so safe for bang to remove unused code.
Will refactor the code to make safety more clear.

File: OptimizelySDK/Data Model/DispatchEvents/EventForDispatch.swift (hunk 4 above)
Comment (jaeopt) [reply OptimizelySDK/Data Model/DispatchEvents/EventForDispatch.swift]: Looks like redundant to "isEqual" above. WIl be removed.

File: OptimizelySDK/OptimizelyTests/OptimizelyTests-DataModel/ConditionLeaf.swift (hunk 5 above)
Comment (jaeopt) [reply OptimizelySDK/OptimizelyTests/OptimizelyTests-DataModel/ConditionLeaf.swift]: Good catch. Looks like an error introduced while merging. Will clean up xcode project.

File: OptimizelySDK/OptimizelyTests/OptimizelyTests-DataModel/ConditionHolderTests_Evaluate.swift [hunk 6]
```diff
@@ -162,3 +177,39 @@ extension ConditionHolderTests_Evaluate {
...
+        let model: ConditionHolder = try! OTUtils.model(from: [["and", userAttributeData]])
+        do {
+            _ = try model.evaluate(project: project, attributes: attributeData)
+            XCTAssert(false, "only operator or leaf node can be the first item")
+        } catch {
+            XCTAssert(true)
+        }
+    }
+    
+    func testEvaluate_OperatorOnly() {
```
Comment (thomaszurkan-optimizely) [inline OptimizelySDK/OptimizelyTests/OptimizelyTests-DataModel/ConditionHolderTests_Evaluate.swift]: Where is the error thrown here?  This appears to now hit the  self[1...] where it would throw an exception

Comment (jaeopt) [reply OptimizelySDK/OptimizelyTests/OptimizelyTests-DataModel/ConditionHolderTests_Evaluate.swift]: I guess you expect out-of-bound exception here. It does not happen since the array size is 1 here ( "and" is not filtered out yet).
